Lib/asyncio/tasks.py

```diff
@@ -74,7 +80,10 @@ def __init__(self, coro, *, loop=None):
         self._fut_waiter = None
         self._must_cancel = False
         self._loop.call_soon(self._step)
-        self.__class__._all_tasks.add(self)
+        try:
+            self._loop._register_task(self)
+        except (AttributeError, NotImplementedError):
+            self.__class__._all_tasks.add(self)
```

Comment:
Let's remove `Task._all_tasks` and `Task._current_tasks`.  They were never public attributes anyways.

Reply:
These attributes are present for sake of backward compatibility.
Let's imagine a loop without task introspection support (not implementing `_register_task`, `_enter_task` and `_leave_task`).
The proposed PR will keep working by `all_tasks`/`_current_task` fallbacks.

Reply:
Got it.  Let's add a comment that this code is needed to thrird-party event loops end that it will be removed in 3.8.